### PR TITLE
chore: slightly improve locking when adding reward vote

### PR DESCRIPTION
**Pull request by MatusKysel**

### CODE REVIEW

File: libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp [hunk 1]
```diff
@@ -479,17 +480,23 @@ uint64_t VoteManager::roundDeterminedFromVotes(size_t two_t_plus_one) {
...
+  }
+
   {
-    std::unique_lock lock(reward_votes_mutex_);
-    if (reward_votes_.contains(vote_hash)) {
-      // Should not happen, we have checked seen votes in VotePacketHandler::process
-      LOG(log_er_) << "Reward vote " << vote_hash << " saved already";
-      return false;
-    }
+    std::shared_lock lock(reward_votes_mutex_);
```
[the remark this answers is not in the document]
Comment (MatusKysel) [reply libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp:499]: nope reward_votes_pbft_block_hash_ need to be proteced

**Review by JakubFornadel — APPROVED** [review]

File: libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (JakubFornadel) [reply libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp:499]: can you pls add comment here that reward_votes_mutex_ protects both reward_votes_pbft_block_hash_ and reward_votes_ ?

File: libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp [hunk 2]
```diff
@@ -479,25 +480,42 @@ uint64_t VoteManager::roundDeterminedFromVotes(size_t two_t_plus_one) {
...
   const auto voted_block_hash = vote->getBlockHash();
+  const auto vote_step = vote->getStep();
+  const auto vote_type = vote->getType();
+
+  if (vote_type != cert_vote_type) {
+    LOG(log_dg_) << "Drop vote " << vote_hash << ". The vote type " << vote_type << " is not cert vote type.";
+    return false;
+  }
+
+  if (vote_step != 3) {
```
Comment (kstdl) [inline libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp:493]: Why is it should be 3? 
I see that it was this way before you changes, but I think we need to improve checks like this somehow. At least with a comment

Comment (MatusKysel) [reply libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp:493]: I dunno why @gaoqi8649  should answer that :)

Comment (kstdl) [reply libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp:493]: https://github.com/Taraxa-project/taraxa-node/issues/1804

**Review by kstdl — APPROVED** [review]

**Review by mfrankovi — APPROVED** [review]

**Pull request merged**